app/model_store.cpp

```
@@ -0,0 +1,1171 @@
+        std::error_code error;
+        fs::remove(curl_errors, error);
+        throw std::runtime_error(
+            "curl failed while downloading " + model.repo + " (exit code " +
+            std::to_string(status) + ")" + (detail.empty() ? "" : ": " + detail));
+    }
+    std::error_code error;
+    fs::remove(curl_errors, error);
+}
+
+bool
+valid_file(const fs::path& path, const Artifact& artifact) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)